Project1/kacsavadaszat/kacsavadaszat.pde

```diff
@@ -0,0 +1,187 @@
+int pontszam, eletek, kacsakAKepernyon, megoltKacsak;
+int maxKacsak = 5;
+
+float[] kacsaX = new float[maxKacsak];
+float[] kacsaY = new float[maxKacsak];
+float[] kacsaSzuletett = new float[maxKacsak];
+float[] kacsaElet = new float[maxKacsak];
+
+float celkeresztMeret, kacsaMeret, talajszint;
+
+PImage talaj, bokor, kacsa;
+
+boolean jatekban;
+
+void setup() {
+  size(800, 600);
+
+  talajszint = height - 205;
+  talaj = loadImage("talaj.png"); // 800x205
+  bokor = loadImage("bokor.png"); // 45x81
+  kacsa = loadImage("kacsa.png"); // 35x31
+
+  megoltKacsak = 0; // Nyomon követi, hogy mennyi kacsa halt meg azóta, hogy hozzáadtunk egy újat.
+  kacsakAKepernyon = 1;
+  kacsaMeret = 35;
+  celkeresztMeret = 22;
+  jatekban = false; // Innen tudjuk, hogy a játékot kell renderelni, vagy a kezdő/vég képernyőt.
+
+  reset();
+}
+
+void draw() {
+  background(#64b1ff);
+
+  if (jatekban) {
+    // Játékban vagyunk és az életek sem fogytak még el.
+    if (eletek > 0) {
+      // Megkeresi a túlélő kacsákat és újrateremti őket.
+      for (int kacsaSorszam = 0; kacsaSorszam < kacsakAKepernyon; ++kacsaSorszam) {
+        if (frameCount - kacsaSzuletett[kacsaSorszam] > kacsaElet[kacsaSorszam] * frameRate) {
+          kacsakAKepernyon = 1;
+          eletek -= 1;
+          megoltKacsak = 0;
+          kacsaTeremt(kacsaSorszam);
+        }
+      }
+
+      // Minden valahány megölt kacsa után egyel többet teremtünk. (5, 15, 30, 50, 75 )
+      if (megoltKacsak > 0 && megoltKacsak % (5 * kacsakAKepernyon) == 0 && kacsakAKepernyon < maxKacsak) {
+        megoltKacsak = 0;
+        kacsakAKepernyon += 1;
+        kacsaTeremt(kacsakAKepernyon - 1);
+      }
+
+      kacsaRajzol();
+    } else {
+      // Elfogytak az életek, megállítjuk a játékot
+      jatekban = false;
+    }
+  } else {
+    // Meg van állítva a játék, kiderítjuk miért
+    if (eletek == 0) {
+      vegeredmeny();
+    } else {
+      kezdokepernyo();
+    }
+
+    // Kattintásra újraindul a játék
+    if (mousePressed) {
+      jatekban = true;
+      reset();
+      kacsaTeremt(0);
+    }
+  }
+
+  // Az eredményjelzőt és a célkeresztet rajzoljuk minden fölé.
+  talajRajzol();
+  eredmenyjelzo();
+  celkeresztRajzol();
+}
+
+void mousePressed() {
```

Comment:
Nem garantaljuk :)
Irtam a readmebe, hogy szorgalminak be lehet tenni pontlevonast ha kattint de nincs talalat, es ha 0 hoz er akkor game over.